### PR TITLE
CMake: add support for pkg-config

### PR DESCRIPTION
**Pull request by mwtoews**
This addresses part of #2546 to install a pkg-config file.

### CODE REVIEW

**Comment by mwtoews** [comment]
Does anyone feel qualified to review this? Nevertheless, the CMake-generated set-up passes post-install tests with Linux and macOS.

One observation discovered here between autotools vs CMake installs on macOS is that the dynamic library via CMake has the relative reference `@rpath/libproj.22.dylib` (via `otool -L`) whereas the autotools library uses an absolute path '/tmp/proj_cmake_install/lib/libproj.22.dylib`. This shouldn't be surprising given the CMake options, but is a difference to note.

**Comment by rouault** [comment]
not a cmake expert, but that looks very reasonable to me

**Comment by dg0yt** [comment]
Given the syntax of the root `proj.pc.in`, did you conside creating a function (i.e. scope) that sets the variables needed to configure that file, instead of introducing a separate `cmake/proj.pc.cmake.in`? But changes should be rare, so redundancy may be acceptable.

File: CMakeLists.txt [hunk 1]
```diff
@@ -282,3 +282,24 @@ endif()
...
+    set(CURL_LIBS -lcurl)
+  endif()
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/proj.pc.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/proj.pc
+    @ONLY)
+
+  install(FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/proj.pc
+    DESTINATION lib/pkgconfig)
```
Comment (dg0yt) [inline CMakeLists.txt]: I guess Debian will be unhappy about this rule. Due to multi-arch, `proj.pc` resides in `/usr/lib/x86_64-linux-gnu/pkgconfig/`. In a `GNUInstallDirs` context, the destination would probably be `${CMAKE_INSTALL_LIBDIR}/pkgconfig`.

**Comment by kbevers** [comment]
Is everyone happy with this PR in its current state?

File: CMakeLists.txt [hunk 2]
```diff
@@ -282,3 +282,24 @@ endif()
 set(docfiles COPYING NEWS AUTHORS)
 install(FILES ${docfiles}
         DESTINATION "${DOCDIR}")
+
+################################################################################
+# pkg-config support
+################################################################################
+if(UNIX)
```
Comment (dg0yt) [inline CMakeLists.txt]: I just had nothing but MinGW (MSYS2) today, and guess what... There are other pkg-config files here. I successfully tested build/install with:
~~~
if(UNIX OR MINGW)
~~~

Comment (mwtoews) [reply CMakeLists.txt]: hmm, possibly this one too, as some of these paths are set from `GNUInstallDirs` https://github.com/OSGeo/PROJ/blob/1a518773616fb001483c175e8dfab00c06e83c89/cmake/ProjInstallPath.cmake#L26-L27

**Comment by mwtoews** [comment]
> Given the syntax of the root `proj.pc.in`, did you conside creating a function (i.e. scope) that sets the variables needed to configure that file

Sure, I'll give this a "go"...

File: CMakeLists.txt (hunk 2 above)
Comment (dg0yt) [reply CMakeLists.txt]: I see. I don't see a difference in `proj.pc` when I change `ProjInstallPath.cmake`.
And I wouldn't delay PROJ 8.0.0 for MINGW pkgconfig.

Default config for MinGW, generator Ninja:
~~~
prefix=c:/OSGeo4W
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include
datarootdir=${prefix}/share
datadir=${prefix}/share/proj

Name: PROJ
Description: Coordinate transformation software library
Requires:
Version: 8.0.0
Libs: -L${libdir} -lproj
Libs.private: -lsqlite3 -ltiff -lcurl -lstdc++
Cflags: -I${includedir}
~~~

Looks good to me.

**Comment by dg0yt** [comment]
LGTM. 
~~~
$ diff former-mingw current-mingw
6c6
< datadir=${prefix}/share/proj
---
> datadir=${datarootdir}/proj

**Comment by mwtoews** [comment]
This last update, which uses the same `proj.pc.in` file as by autotools, creates an identical `proj.pc` file for a given prefix.

**Pull request merged**